src/main/java/org/jenkinsci/plugins/github_branch_source/AbstractGitHubSCMSource.java

```diff
@@ -250,11 +250,22 @@ public String getRemote() {
     }
 
     @Override
+    @CheckForNull
     protected SCMRevision retrieve(SCMHead head, TaskListener listener) throws IOException, InterruptedException {
-        // TODO reimplement using GitHub API
-        return super.retrieve(head, listener);
+        StandardCredentials credentials = Connector.lookupScanCredentials(getOwner(), apiUri, scanCredentialsId);
+        if (credentials == null) {
+            listener.getLogger().println("No scan credentials, skipping");
+            return null;
+        }
+        listener.getLogger().format("Connecting to %s using %s%n", getDescriptor().getDisplayName(), CredentialsNameProvider.name(credentials));
+        GitHub github = Connector.connect(apiUri, credentials);
```

Comment:
@KostyaSha This plugin manages its own credentials, you can find [other example](https://github.com/jenkinsci/github-branch-source-plugin/blob/master/src/main/java/org/jenkinsci/plugins/github_branch_source/AbstractGitHubSCMSource.java#L194). Additionally, we are using only `https` because the GH API is exposed through `https`.